skyportal/handlers/api/observation_plan.py

```diff
@@ -0,0 +1,150 @@
+import jsonschema
+from marshmallow.exceptions import ValidationError
+
+from baselayer.app.access import permissions
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    ObservationPlanRequest,
+    Group,
+    Allocation,
+)
+
+from ...models.schema import ObservationPlanPost
+
+
+class ObservationPlanRequestHandler(BaseHandler):
+    @permissions(["Upload data"])
+    def post(self):
+        """
+        ---
+        description: Submit observation plan request.
+        tags:
+          - observationplan_requests
+        requestBody:
+          content:
+            application/json:
+              schema: ObservationPlanPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New observation plan request ID
+        """
+        data = self.get_json()
+
+        try:
+            data = ObservationPlanPost.load(data)
+        except ValidationError as e:
+            return self.error(
+                f'Invalid / missing parameters: {e.normalized_messages()}'
+            )
+
+        data["requester_id"] = self.associated_user_object.id
+        data["last_modified_by_id"] = self.associated_user_object.id
+        data['allocation_id'] = int(data['allocation_id'])
+        data['localization_id'] = int(data['localization_id'])
+
+        allocation = Allocation.get_if_accessible_by(
+            data['allocation_id'],
+            self.current_user,
+            raise_if_none=True,
+        )
+
+        instrument = allocation.instrument
```

Comment:
Just writing this out here.
Allocations: I have time on an Instrument, and therefore have the power to submit requests to it.
ObservationPlanRequests: For an Instrument I have an Allocation on, I am requesting, for a specific set of parameters as given by the front-end page, a schedule appropriate for that telescope (and GCNEvent).
ObservationPlans: The resulting observation plan from that ObservationPlanRequest.
This is mostly for the developers, because the user does not care about this subtlety, so not quite sure where to put it?

Reply:
Maybe add a file in the docs for TOO / MMA followup. Explain everything for dummies that don't know what these terms mean, give an example for following up a GW signal or something. 
There's also some subtleties in where the ObservationPlans are both database entries but also get sent out to instruments via their APIs (they're not just data in a table). 
You've obviously thought a lot about this but other users will not easily understand what all these tools are used for.

Reply:
I added some doc stuff!